src/main/resources/META-INF/plexus/components.xml

```diff
@@ -17,6 +22,7 @@
        <package>org.apache.maven.plugins:maven-jar-plugin:2.6:jar,${project.groupId}:${project.artifactId}:${project.version}:build</package>
        <pre-integration-test>${project.groupId}:${project.artifactId}:${project.version}:start</pre-integration-test>
        <integration-test>org.apache.maven.plugins:maven-failsafe-plugin:2.19.1:integration-test</integration-test>
+       <!-- post-integration-test is opportunity for code coverage, such as jacoco, to collect statistics from running container -->
        <post-integration-test></post-integration-test>
        <verify>${project.groupId}:${project.artifactId}:${project.version}:stop,org.apache.maven.plugins:maven-failsafe-plugin:2.19.1:verify</verify>
```

Comment:
It's documented that order of plugins is [those specified by packaging](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#Plugins), followed by [order declared in pom](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#A_Build_Phase_is_Made_Up_of_Plugin_Goals).

[Maven Failsafe](https://maven.apache.org/surefire/maven-failsafe-plugin/), usually binds with _integration-test_ and _verify_.  [Jacoco report-integration-mojo](http://eclemma.org/jacoco/trunk/doc/report-integration-mojo.html) likewise binds to _verify_.  So, current practice is that running and actually verifying integration tests requires using _verify_ phase.

I was guided in my choices by what other container technologies use; [Cargo](https://codehaus-cargo.github.io/cargo/Maven2+Plugin+Tips.html) recommends associating start goal with _pre-integration-test_ and stop goal with _post-integration-test_. [Tomcat maven plugin](http://tomcat.apache.org/maven-plugin-2.2/run-mojo-features.html) recommends associating run-war-only goal with _pre-integration-test_ and shutdown goal with _post-integration-test_.

I would prefer not to bind [jacoco:dump](http://eclemma.org/jacoco/trunk/doc/dump-mojo.html) with integration-test.  This goal must be called after failsafe runs the tests and before the container is stopped.  It's often difficult to ensure the ordering of goals within a phase, particularly when parent poms and profiles are involved.  You already understand the problem with binding to post-integration-test.